spec/controllers/submissions_controller_spec.rb

```diff
@@ -998,6 +1019,23 @@
             expect(response).to render_template(partial: 'shared/_poll_job')
           end
 
+          it 'should succeed if it is before the global due date but collect_current is true' do
+            @assignment.update!(due_date: 1.week.from_now)
+            allow(Assignment).to receive_message_chain(
+              :includes, :find
+            ) { @assignment }
+            expect_any_instance_of(SubmissionsController).not_to receive(:flash_now).with(:error, anything)
```

Comment:
same here, can we split this up into two tests and test for the presence of an error message in the flash hash instead?